### DISCUSSION TITLE
How to plot schmid factor into a pole figure

### DISCUSSION THREAD

**Discussion by Brian-Liu-rx** [mtex-toolbox/mtex#1049] [Ask Anything]
Hi,
    I am looking for a way to plot  schmid factor colormap into a pole figure, just like the below.
    What I know is to firstly plot a pole figure and  "hold on", then plot schmid factor colormap on it, but I don't know how to plot SF.
    Thanks!
![image](https://user-images.githubusercontent.com/69452222/119218574-9bf17f00-bb13-11eb-8956-8c96d48124e9.png)

**Comment by ralfHielscher** [comment]
Hi Brian,

can you provide a formula for what you want to plot. In my understanding 
the Schmid factor is (ori*b·t)*(ori*n·t) where b is the Burgers vector, n is the normal directions of the slip system, and t is the tension direction. This gives us a value for each orientation ori. How to make from this a value for a point r in an (hkl) pole figure?

Ralf.

**Comment by QPF-BS** [comment]
Mr. Ralf Hielscher has provided a code that might solve this problem. The largest Schmid factor would be plotted Schmid factor colormap into a pole figure. What I know is how to plot the largest Schmid factor into an inverse pole figure？
![image](https://user-images.githubusercontent.com/84628290/119228196-79c42500-bb44-11eb-8d84-13fb0e8b2943.png)

**Reply by QPF-BS** [reply]
What I want to know is how to plot the largest Schmid factor into an inverse pole figure？

**Reply by ralfHielscher** [reply]
Maybe we could avoid the terms _pole figure_ and _inverse pole figure_. They only cause confusion. The essence of plotting something in a pole figure is visualize it with respect to the specimen reference frame, i.e., way visualize for a certain crystal orientation the alignment of a certain of one of its crystal planes in a pole figure. Accordingly, in an inverse pole figure something is visualized with respect to crystal coordinates.

Now you aim at visualizing the maximum Schmidt factor of a certain family of slip systems as a function of some direction. Which direction: the burgers vector, the normal direction, the tension direction (this what is shown in the documentation).

You should explain very clearly what you want to visualize.

Ralf.

**Reply by Brian-Liu-rx** [reply]
Exactly，ralf，What we want to do is to visualize the maximum Schmidt factor of a certain family of slip systems as a function of some direction.   

I try the method QPF-BS just mentioned, but I think that code given by help mannul is to plot the  maximum SF of various load direction with respect to crystal orientation, not to plot the  maximum SF of various crystal orientation with respect to one centain load direction.  Something similar with PF and IPF.

**Reply by ralfHielscher** [reply]
The point I want to make is 
(1) you can assign to an orientation a Schmid factor
(2) you can assign to an orientation a position in a pole figure
However, as the assignment (2) is not one to one you can not assign a position in a pole figure a Schmid factor. At least you need to specify for a certain pole figure position which orientation to choose for the computation of the Schmid factor.

Ralf.

**Reply by QPF-BS** [reply]
Sir, in fact, I want to draw a figure as shown in the figure below. The maximum Schmidt factor of different orientations was drawed in an IPF.
![图片1](https://user-images.githubusercontent.com/84628290/119231367-1c83a000-bb53-11eb-89fb-c15ae145f3fb.jpg)

**Reply by ralfHielscher** [reply]
Please try harder to specify exactly what you want to draw. At the end you want to use the picture in a paper and you need to specify anyway what it represents.  The sentence "The maximum Schmidt factor of different orientations was drawed in an IPF" is quite unsatisfactory. If you would better understand what you want to draw it would be also more easy for you to draw it.

Ralf.

**Reply by QPF-BS** [reply]
Taking FCC metal as an example, the slip system with the largest Schmidt factor starts first and dominates the deformation process. Therefore, the maximum Schmid factor of the alloy with different orientations can be expressed to determine which orientation is more prone to slip. As shown in the figure above，the[1 1 8.22] orientation had the biggest Schmid factor o.5 and the [111] had a smaller. hence, the[1 1 8.22] orientation alloy would be easier slid during deformation

**Reply by ralfHielscher** [reply]
What you are missing is that [1 1 8.22] is NOT an orientation but a crystal direction. An orientation describes the alignment of a crystal within your specimen and is commonly described by three Euler angles. 
Here you have three Miller indices [h,k,l]. Please explain which crystal orientation corresponds to these Miller indices!

Ralf.

**Reply by QPF-BS** [reply]
Sir, for BCC and FCC, [1 1 8.22] was the crystal direction，and also was the stress orientation

**Reply by King-wang-H** [reply]
Hi Brian,
I am also learning to draw such a diagram recently. Have you solved your problem?
I wonder if it is convenient for you to have your advice.
